tests/engine/test_calc_job.py

```diff
@@ -354,62 +354,106 @@ def test_parse_retrieved_folder(self):
         assert exit_code == process.exit_codes.ERROR_READING_OUTPUT_FILE
 
-    def test_parse_insufficient_data(self):
-        """Test the scheduler output parsing logic in `CalcJob.parse`.
 
-        Here we check explicitly that the parsing does not except even if the required information is not available.
-        """
-        process = self.instantiate_process()
-        retrieved = orm.FolderData().store()
-        retrieved.add_incoming(process.node, link_label='retrieved', link_type=LinkType.CREATE)
-        process.parse()
+@pytest.fixture
+def process(aiida_local_code_factory):
+    """Instantiate a process with default inputs and return the `Process` instance."""
+    from aiida.engine.utils import instantiate_process
+    from aiida.manage.manager import get_manager
 
-        filename_stderr = process.node.get_option('scheduler_stderr')
-        filename_stdout = process.node.get_option('scheduler_stdout')
+    inputs = {
+        'code': aiida_local_code_factory('arithmetic.add', '/bin/bash'),
+        'x': orm.Int(1),
+        'y': orm.Int(2),
+        'metadata': {
+            'options': {}
+        }
+    }
 
-        # The scheduler parsing requires three resources of information, the `detailed_job_info` dictionary which is
-        # stored as an attribute on the calculation job node and the output of the stdout and stderr which are both
-        # stored in the repository. In this test, we haven't created these on purpose. This should not except the
-        # process but should log a warning, so here we check that those expected warnings are attached to the node
-        logs = [log.message for log in orm.Log.objects.get_logs_for(process.node)]
-        expected_logs = [
-            'could not parse scheduler output: the `detailed_job_info` attribute is missing',
-            'could not parse scheduler output: the `{}` file is missing'.format(filename_stderr),
-            'could not parse scheduler output: the `{}` file is missing'.format(filename_stdout)
-        ]
+    manager = get_manager()
+    runner = manager.get_runner()
 
-        for log in expected_logs:
-            assert log in logs
+    process_class = CalculationFactory('arithmetic.add')
+    process = instantiate_process(runner, process_class, **inputs)
+    process.node.set_state(CalcJobState.PARSING)
 
-    def test_parse_not_implemented(self):
-        """Test the scheduler output parsing logic in `CalcJob.parse`.
+    return process
 
-        Here we check explicitly that the parsing does not except even if the scheduler does not implement the method.
-        """
-        process = self.instantiate_process()
-        retrieved = orm.FolderData().store()
-        retrieved.add_incoming(process.node, link_label='retrieved', link_type=LinkType.CREATE)
 
-        process.node.set_attribute('detailed_job_info', {})
+@pytest.mark.usefixtures('clear_database_before_test', 'override_logging')
+def test_parse_insufficient_data(process):
```

Comment:
So, this is maybe separate from this PR but just to learn: what is the difference between passing a fixture via `@pytest.mark.usefixtures` and passing it as a parameter of the test?

Reply:
Pass it as function argument if the fixture actually needs to be called in the test body, otherwise use it in `pytest.mark.usefixtures`. The reason is simply that, although the `pytest` magic of automatically injecting fixtures into a test by defining them as an argument, it leads to many unused function arguments which is weird to read, and leads to `pylint` complaining, for the very same reason.